monai/transforms/lazy/utils.py

```diff
@@ -38,7 +38,7 @@ def is_affine_shaped(data):
             return False
         if not hasattr(data, "shape") or len(data.shape) < 2:
             return False
-        return data.shape[-1] in (3, 4) and data.shape[-2] in (3, 4) and data.shape[-1] == data.shape[-2]
+        return data.shape[-1] in (3, 4) and data.shape[-1] == data.shape[-2]
```

Comment:
This fix is good for now for homogeneous coordinates, in the future we may change to include transforms in shape 2x3 or 3x4